src/Microsoft.FeatureManagement/FeatureDefinition.cs

```diff
@@ -42,5 +42,25 @@ public class FeatureDefinition
         /// A list of variant definitions that specify a configuration to return when assigned.
         /// </summary>
         public IEnumerable<VariantDefinition> Variants { get; set; } = Enumerable.Empty<VariantDefinition>();
+
+        /// <summary>
+        /// Metadata that can be used to group feature flags.
+        /// </summary>
+        public string Label { get; set; }
```

Comment:
This is more of an open discussion.

The Label, ETag, and Tags are non-essential properties of a feature flag definition, meaning they don't participate in feature flag evaluation. Their only usage is that they could be included in telemetry if a publisher chooses to.  This is not obvious from the way they are defined today. I agree we should include them, but I'm wondering if there is a better way to include them.

Reply:
Makes me think if all of the add on metadata should be in some field named TelemetryMetadata. Then people could put whatever they want. No special casing any scenario. Publisher sends all the telemetry metadata as custom properties.

Reply:
> This is more of an open discussion.

Oh, it made enough sense to me to describe it as "metadata" until it's used, so I did that. 

>  in some field named TelemetryMetadata

I'm worried that TelemeteryMetadata might put all of these in a box that they might not belong in. Label is a good example. If we see customers using multiple labels in a single deployment. We might end up supporting a IsEnabledAsync(flag, label). Maybe just a more generic "Metadata" field..?

Reply:
The intent of TelemetryMetadata would be to flow things from a feature definition into telemetry. Right now that's all `Label` is. If we make such a change as you suggest we would merely stop putting it in the TelemetryMetadata bag, add it as a real property on FeatureDefinition because it would have a use, and telemetry publisher would publish it because it's a relevant piece of the feature definition.

Reply:
Oh that's reasonable. I was thinking it would be a breaking change to move it, but that makes sense.

Reply:
I like the telemetry bag idea. It further opens up this library as a platform to support richer telemetry regardless of feature flag sources. AppConfig provider will be a customer benefiting from this extensibility.

I found I had a hard time explaining what Label, ETag, and Tags are to a feature flag without mentioning telemetry or bringing in the concept of AppConfig. It seems all of them should go to the telemetry bag.

Reply:
Added TelemetryMetadata. I kept TelemetryEnabled outside of it. Let me know if you disagree.

Reply:
> Added TelemetryMetadata. I kept TelemetryEnabled outside of it. Let me know if you disagree.

Right, TelemetryEnabled is a functional property of the flag. Not even meant to be sent to the telemetry destination.